src/main/java/com/impactupgrade/nucleus/client/SfdcClient.java

```diff
@@ -523,19 +524,25 @@ protected QueryResult queryEmailContacts(String updatedSinceClause, String filte
     // IMPORTANT: If env.json defines emailOptOut/emailBounced, also include those contacts in this query! This might seem backwards,
     // but we need them in the results so that we can archive them in Mailchimp.
     List<String> clauses = new ArrayList<>();
-    if (!Strings.isNullOrEmpty(env.getConfig().salesforce.fieldDefinitions.emailOptIn)
-        && !env.getConfig().salesforce.fieldDefinitions.listFilterOverridesOptIn) {
-      clauses.add(env.getConfig().salesforce.fieldDefinitions.emailOptIn + "=TRUE");
+    EnvironmentConfig.CRMFieldDefinitions crmFields = env.getConfig().salesforce.fieldDefinitions;
+
+    if (!Strings.isNullOrEmpty(crmFields.emailOptIn) && !crmFields.listFilterOverridesOptIn) {
+      clauses.add(crmFields.emailOptIn + "=TRUE");
 
       // ONLY ADD THESE IF WE'RE INCLUDING THE ABOVE OPT-IN FILTER! Otherwise, some orgs only have opt-out defined,
       // and we'd effectively be syncing ONLY unsubscribes.
-      if (!Strings.isNullOrEmpty(env.getConfig().salesforce.fieldDefinitions.emailOptOut)) {
-        clauses.add(env.getConfig().salesforce.fieldDefinitions.emailOptOut + "=TRUE");
+      if (!Strings.isNullOrEmpty(crmFields.emailOptOut)) {
+        clauses.add(crmFields.emailOptOut + "=TRUE");
       }
-      if (!Strings.isNullOrEmpty(env.getConfig().salesforce.fieldDefinitions.emailBounced)) {
-        clauses.add(env.getConfig().salesforce.fieldDefinitions.emailBounced + "=TRUE");
+      if (!Strings.isNullOrEmpty(crmFields.emailBounced)) {
+        clauses.add(crmFields.emailBounced + "=TRUE");
       }
     }
+
+    if (!Strings.isNullOrEmpty(crmFields.smsOptIn) && !crmFields.listFilterOverridesOptIn) {
+      clauses.add(crmFields.smsOptIn + "=TRUE");
+    }
```

Comment:
@VSydor I **think** we should probably remove this block for now. IIUC, DR wants SMS to be additive to contacts that are already in MC due to an email subscription, and no contacts in MC purely for SMS (and no email). And if you look at the uses of the `CrmContact.canReceiveEmail()` method, you'll see we're using it to decide multiple pieces of the sync, including who to archive. Maybe in the future they'll want us to sync SMS-only contacts. But for now, kill this block and we'll still sync only contacts that are email-ready? And if they happen to also be SMS opted in, that gets set too.

Reply:
You're right
I was overthinking this (thought should be very separate filter)